uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -847,6 +847,76 @@ public function testFindOverlappingLeaveRequestsForOneOverlappingLeaveRequest()
     $this->assertEquals($leaveRequest1->id, $overlappingRequests[0]->id);
   }
 
+  public function testManagerCanCancelOrRejectUserLeaveRequestEvenIfBalanceChangeIsGreaterThanPeriodEntitlementBalanceChangeWhenAllowOveruseFalse() {
+    $manager = ContactFabricator::fabricate();
+    $staff = ContactFabricator::fabricate();
+    $entitlementFromDate = CRM_Utils_Date::processDate('2016-01-01');
+    $entitlementToDate = CRM_Utils_Date::processDate('2016-12-31');
+    $requestFromDate = new DateTime('2016-06-14');
+    $requestToDate = new DateTime('2016-06-14');
+    $requestFromDateType = $this->leaveRequestDayTypes['all_day']['value'];
+    $requestToDateType = $this->leaveRequestDayTypes['all_day']['value'];
+
+    $this->registerCurrentLoggedInContactInSession($manager['id']);
+    $this->setContactAsLeaveApproverOf($manager, $staff, null, null, true, 'has Leaves Approved By');
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => $entitlementFromDate,
+      'end_date'   => $entitlementToDate,
+    ]);
+
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'allow_overuse' => 0
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $staff['id'],
+      'period_id' => $absencePeriod->id
+    ]);
+
+    $entitlementBalanceChange = 0;
+    $this->createLeaveBalanceChange($periodEntitlement->id, $entitlementBalanceChange);
+    $periodStartDate = $absencePeriod->start_date;
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      ['period_start_date' => $periodStartDate]
+    );
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $periodEntitlement->contact_id,
+      'pattern_id' => $workPattern->id
+    ]);
```

Comment:
No need to assign a work pattern to contact. 
Just this is enough: `$workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => 1);`

Reply:
Done.